File: app/src/main/java/alireza/nezami/dependency_injection_without_dagger/java/approach1/ui/MainActivity.java
package alireza.nezami.dependency_injection_without_dagger.java.approach1.ui;

import android.os.Bundle;
import android.widget.TextView;



import androidx.appcompat.app.AppCompatActivity;

import alireza.nezami.dependency_injection_without_dagger.R;

public class MainActivity extends AppCompatActivity {

    public MainViewModel viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // initialize MainViewModel
        viewModel = new MainViewModel(this.getApplicationContext());

        TextView tvData = findViewById(R.id.tvData);
        tvData.setText(viewModel.getSomeData());
    }
}

File: app/src/main/java/alireza/nezami/dependency_injection_without_dagger/kotlin/approach1/ui/MainActivity.kt
package alireza.nezami.dependency_injection_without_dagger.kotlin.approach1.ui

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import alireza.nezami.dependency_injection_without_dagger.R
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    private lateinit var viewModel: MainViewModel
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // initialize MainViewModel
        viewModel = MainViewModel(this.applicationContext)
        val tvData = findViewById<TextView>(R.id.tvData)
        tvData.text = viewModel.someData
    }
}